pyuvdata/utils.py

```diff
@@ -959,3 +959,123 @@ def _reraise_context(fmt, *args):
         ex.args = (cstr, ) + ex.args[1:]
 
     raise
+
+
+def collapse(a, alg, weights=None, axis=None, returned=False):
+    ''' Parent function to collapse an array with a given algorithm.
+    Args:
+        a (array): Input array to process.
+        alg (str): Algorithm to use. Must be defined in this function with
+            corresponding subfunction below.
+        weights (array, optional): weights for collapse operation (e.g. weighted mean).
+            NOTE: Some subfunctions do not use the weights. See corresponding doc strings.
+        axis (int, tuple, optional): Axis or axes to collapse. Default is all.
+        returned (Bool): Whether to return sum of weights. Default is False.
+    '''
+    if alg == 'mean':
```

Comment:
Might be cleaner to define a dict with keys equal to the algorithm names and values equal to the functions. Then you can use `getattr` to call the appropriate function. For a similar example, see the `interp` method on UVBeam.